annotated.go

```diff
@@ -250,23 +250,14 @@ func (la *lifecycleHookAnnotation) apply(ann *annotated) error {
 		)
 	}
 
-	if ft.NumIn() < 1 || ft.In(0) != _typeOfContext {
-		return fmt.Errorf(
-			"first argument of hook must be context.Context, got %v (%T)",
-			la.Target,
-			la.Target,
-		)
-	}
-
-	hasOut := ft.NumOut() == 1
-	returnsErr := hasOut && ft.Out(0) == _typeOfError
-
-	if !hasOut || !returnsErr {
-		return fmt.Errorf(
-			"hooks must return only an error type, got %v (%T)",
-			la.Target,
-			la.Target,
-		)
+	if n := ft.NumOut(); n > 0 {
+		if n > 1 || ft.Out(0) != _typeOfError {
+			return fmt.Errorf(
+				"optional hook return may only be error, got %v (%T)",
```

Comment:
We may want to specify that the optional return is `an` error to avoid confusions between a single error vs. multiple errors.

```suggestion
				"optional hook return may only be an error, got %v (%T)",
```

Reply:
Done, thanks!